test/libyul/yulOptimizerTests/ssaAndBack/for_loop.yul

```diff
@@ -1,7 +1,7 @@
 {
+  let a := mload(0)
```

Comment:
Similarly here.

Reply:
@chriseth  on running this test(after adding pre blocks in loop) i am getting this error, does that mean ssaBack tests are run through `RedundantAssignEliminator` first?
```
/home/fnatic/solidity/test/boostTest.cpp(116): error: in "yulOptimizerTests/ssaAndBack/for_loop": Exception during extracted test: /home/fnatic/solidity/libyul/optimiser/RedundantAssignEliminator.cpp(123): Throw in function virtual void yul::RedundantAssignEliminator::operator()(const yul::ForLoop&)
```

Reply:
You are right, these tests have to be modified!